node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala

```diff
@@ -1,171 +1,261 @@
+      implicit val lightBlockEncoder = jsonEncoderOf[F, LightBlockInfo]
+      implicit val lightBlockListEnc = jsonEncoderOf[F, List[LightBlockInfo]]
+      implicit val dataRespEncoder   = jsonEncoderOf[F, DataResponse]
+      implicit val prepareEncoder    = jsonEncoderOf[F, PrepareResponse]
+      implicit val explRespEncoder   = jsonEncoderOf[F, ExploratoryDeployResponse]
+      // Decoders
+      implicit val deployRequestDecoder        = jsonOf[F, DeployRequest]
+      implicit val dataRequestDecoder          = jsonOf[F, DataRequest]
+      implicit val prepareDecoder              = jsonOf[F, PrepareRequest]
+      implicit val exploreDeployRequestDecoder = jsonOf[F, ExploreDeployRequest]
+      implicit val stringDecoder               = jsonOf[F, String]
+      // uri path parser
+      implicit val stringParser = StringParser.strParser[F]
+      implicit val intParser    = StringParser.intParser[F]
 
-    // TODO: Create generic encoders/decoders for
-    // ADT's with discriminator field
-    implicit val encodeByteString: Encoder[ByteString] =
-      Encoder.encodeString.contramap[ByteString](PrettyPrinter.buildStringNoLimit)
-    implicit val encodeLightBlockInfo: Encoder[LightBlockInfo] = deriveEncoder[LightBlockInfo]
-
-    implicit val encodeBlockInfo: Encoder[BlockInfo] = deriveEncoder[BlockInfo]
-    // Encoders
-    implicit val stringEncoder     = jsonEncoderOf[F, String]
-    implicit val booleanEncode     = jsonEncoderOf[F, Boolean]
-    implicit val apiStatusEncoder  = jsonEncoderOf[F, ApiStatus]
-    implicit val blockInfoEncoder  = jsonEncoderOf[F, BlockInfo]
-    implicit val lightBlockEncoder = jsonEncoderOf[F, LightBlockInfo]
-    implicit val lightBlockListEnc = jsonEncoderOf[F, List[LightBlockInfo]]
-    implicit val dataRespEncoder   = jsonEncoderOf[F, DataResponse]
-    implicit val prepareEncoder    = jsonEncoderOf[F, PrepareResponse]
-    implicit val explRespEncoder   = jsonEncoderOf[F, ExploratoryDeployResponse]
-    // Decoders
-    implicit val deployRequestDecoder = jsonOf[F, DeployRequest]
-    implicit val dataRequestDecoder   = jsonOf[F, DataRequest]
-    implicit val prepareDecoder       = jsonOf[F, PrepareRequest]
-    implicit val ExploreDeployRequest = jsonOf[F, ExploreDeployRequest]
-
-    HttpRoutes.of[F] {
-      case GET -> Root / "status" =>
+      def getErrorMsg(ex: Throwable): String =
+        if (ex.getMessage == null) ex.toString else ex.getMessage
+      def handleError(err: Throwable) = err match {
+        // The place where all API errors are handled
+        // TODO: introduce error codes
+        case err: InvalidMessageBodyFailure =>
+          // Request JSON parse errors
+          s"${getErrorMsg(err).take(250)}..."
+        // Errors from BlockAPI
+        case err: BlockApiException =>
+          getErrorMsg(err)
+        case err: Throwable =>
+          // Logging only unanticipated errors, not related to Block API or input parsing (user errors)
+          getErrorMsg(err)
```

Comment:
![image](https://user-images.githubusercontent.com/13496262/97516573-5aaa4e00-19ce-11eb-8180-845ecd661f05.png)

Response Header is : application/json

![image](https://user-images.githubusercontent.com/13496262/97516632-7f9ec100-19ce-11eb-998c-b0bdc27acbf9.png)

The body content is like "<error msg>".

That's what you ask, right?

Reply:
> > > As I mentioned Sep 22, the RhoExpr stuff in the generated swagger.json looks wrong. ...
> > > I tried to fix this but it turned out to be very easy now. I would like to take this version for now and fix it in another pr.
> 
> I don't understand the value of that. Why merge a PR that is clearly incorrect? What is the value of such a swagger.json document to a client developer?
> 
> Perhaps there is some context that I am missing. Perhaps other reviewers are aware of some value that I don't see.

@zsluedem we need to fix generated schema to be in sync with runtime JSON types. Can this library have connection to codec that is used or this is done manually?

I don't like default behavior how scodec serializes to JSON, I'm more for `{ type: "ExprString": ... }`, it would be nice to change this in the future.

Reply:
Yes, this is what I meant. Thanks!